### PR TITLE
Fix golint issues in pkg/apis/apps and pkg/apis/apps/validation

### PR DESCRIPTION
**Pull request by obitech**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix golint issues in `pkg/apis/apps` and `pkg/apis/apps/validation`

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/apis/apps/types.go [hunk 1]
```diff
@@ -261,7 +262,7 @@ type ControllerRevisionList struct {
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (liggitt) [inline pkg/apis/apps/types.go]: keep this comment separated by a space so it doesn't show up in godoc

File: pkg/apis/apps/types.go [hunk 2]
```diff
@@ -343,6 +345,7 @@ type DeploymentRollback struct {
 }
 
 // DEPRECATED.
+// RollbackConfig specifies the state of a revision to roll back to.
```
Comment (liggitt) [inline pkg/apis/apps/types.go]: move this above the DEPRECATED line

File: pkg/apis/apps/types.go [hunk 3]
```diff
@@ -480,7 +488,7 @@ type DeploymentCondition struct {
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
```
Comment (liggitt) [inline pkg/apis/apps/types.go]: keep the extra blank space

File: pkg/apis/apps/types.go (hunk 1 above)
Comment (obitech) [reply pkg/apis/apps/types.go]: Fixed

File: pkg/apis/apps/types.go (hunk 2 above)
Comment (obitech) [reply pkg/apis/apps/types.go]: Fixed

File: pkg/apis/apps/types.go (hunk 3 above)
Comment (obitech) [reply pkg/apis/apps/types.go]: Fixed

**Pull request merged**